pkg/snowflake/materialized_view.go

```diff
@@ -0,0 +1,178 @@
+package snowflake
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/jmoiron/sqlx"
+)
+
+// MaterializedViewBuilder abstracts the creation of SQL queries for a Snowflake Materialized View
+type MaterializedViewBuilder struct {
+	name      string
+	db        string
+	schema    string
+	secure    bool
+	replace   bool
+	comment   string
+	statement string
+}
+
+// QualifiedName prepends the db and schema if set and escapes everything nicely
+func (vb *MaterializedViewBuilder) QualifiedName() string {
+	var n strings.Builder
+
+	if vb.db != "" && vb.schema != "" {
+		n.WriteString(fmt.Sprintf(`"%v"."%v".`, vb.db, vb.schema))
+	}
+
+	if vb.db != "" && vb.schema == "" {
+		n.WriteString(fmt.Sprintf(`"%v"..`, vb.db))
+	}
+
+	if vb.db == "" && vb.schema != "" {
+		n.WriteString(fmt.Sprintf(`"%v".`, vb.schema))
+	}
+
+	n.WriteString(fmt.Sprintf(`"%v"`, vb.name))
+
+	return n.String()
+}
+
+// WithComment adds a comment to the MaterializedViewBuilder
+func (vb *MaterializedViewBuilder) WithComment(c string) *MaterializedViewBuilder {
+	vb.comment = c
+	return vb
+}
+
+// WithDB adds the name of the database to the MaterializedViewBuilder
+func (vb *MaterializedViewBuilder) WithDB(db string) *MaterializedViewBuilder {
+	vb.db = db
+	return vb
+}
+
+// WithReplace adds the "OR REPLACE" option to the MaterializedViewBuilder
+func (vb *MaterializedViewBuilder) WithReplace() *MaterializedViewBuilder {
+	vb.replace = true
+	return vb
+}
+
+// WithSchema adds the name of the schema to the MaterializedViewBuilder
+func (vb *MaterializedViewBuilder) WithSchema(s string) *MaterializedViewBuilder {
+	vb.schema = s
+	return vb
+}
+
+// WithSecure sets the secure boolean to true
+// [Snowflake Reference](https://docs.snowflake.net/manuals/user-guide/views-secure.html)
+func (vb *MaterializedViewBuilder) WithSecure() *MaterializedViewBuilder {
+	vb.secure = true
+	return vb
+}
+
+// WithStatement adds the SQL statement to be used for the view
+func (vb *MaterializedViewBuilder) WithStatement(s string) *MaterializedViewBuilder {
+	vb.statement = s
+	return vb
+}
+
+// View returns a pointer to a Builder that abstracts the DDL operations for a view.
+//
+// Supported DDL operations are:
+//   - CREATE MATERIALIZED VIEW
+//   - ALTER MATERIALIZED VIEW
+//   - DROP MATERIALIZED VIEW
+//   - SHOW MATERIALIZED VIEWS
+//   - DESCRIBE MATERIALIZED VIEW
+//
+// [Snowflake Reference](https://docs.snowflake.com/en/sql-reference/ddl-table.html#materialized-view-management)
+func MaterializedView(name string) *MaterializedViewBuilder {
+	return &MaterializedViewBuilder{
+		name: name,
+	}
+}
+
+// Create returns the SQL query that will create a new view.
+func (vb *MaterializedViewBuilder) Create() string {
```

Comment:
You'll need to include a `USE WAREHOUSE 'FOO';` prefixed before the `CREATE` statement. You won't need it for any other operation. You'll want to still require the `warehouse` attribute on the resource so that you can get the warehouse name.

Reply:
ok

Reply:
It will cause this error.
```
materialized_view_acceptance_test.go:18: Step 1/1 error: Error running apply: 2021/02/18 06:01:58 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.10.0
        
        Error: error creating view JXYUVKLPHP: 000008 (0A000): Actual statement count 2 did not match the desired statement count 1.
```

See here for more about multi statements.
https://docs.snowflake.com/en/user-guide/jdbc-using.html#sending-multiple-statements-handling-results

Reply:
Could you try using our MultiExec functionality? https://github.com/chanzuckerberg/terraform-provider-snowflake/pull/435/files#diff-5c55ac48e02fe8250a8bdbbd86c3a30d903cdf60347f151cc3b0d087ff04f732R17